hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1056,33 +1056,40 @@ int SaraNcpClient::selectSimCard(ModemState& state) {
 
     if (ncpId() == PLATFORM_NCP_SARA_R410) {
         int resetCount = 0;
+        const int imsiMaxRetryCount = 5;
+        const int CcidRetryCount = 2;
         // Note: Not failing on AT error on ICCID/IMSI lookup since SIMs have shown strange edge cases
         // where they error for no reason, and hard resetting the modem or power cycling would not clear it.
         //
         // Check if we are using a Twilio Super SIM based on ICCID
         char buf[32] = {};
-        int ccidCount = 0;
+        int ccidCount = 1;
         do {
             memset(buf, 0, sizeof(buf));
             auto lenCcid = getIccidImpl(buf, sizeof(buf));
             if (lenCcid > 5) {
                 netConf_ = networkConfigForIccid(buf, lenCcid);
                 break;
             }
-        } while (++ccidCount < 2);
+        } while (++ccidCount <= CcidRetryCount);
         // If failed above i.e., netConf_ is still not valid, look for network settings based on IMSI
         if (!netConf_.isValid()) {
-            int imsiCount = 0;
+            int imsiCount = 1;
             do {
+                HAL_Delay_Milliseconds(100*imsiCount);
```

Comment:
It was highly varied in my testing. Sometimes 100ms worked, but most of the times it took a second try. 
I believe 250ms with 3 times is optimal perhaps?... I will add all the other changes. :) thanks!

Reply:
That sounds ok, and if this only happens when UMNOPROF=0 the first time or gets switched to =0 we could add `if (imsiCount > 1) { // delay }` to prevent the delay from happening every time.  It is only 250ms, but they all add up.

Reply:
Good point. Changed! 👍